### ISSUE TITLE
Allow just 'height' to be used for resizing images

### ISSUE THREAD

**Issue by ghost** [backdrop/backdrop-issues#3599]
When you upload an image using the editor, you can specify width and height to resize it. If you just enter a width, the height is calculated automatically based on the aspect ratio of the image. However if you just enter a height, nothing happens - the image stays full-size.

It'd be nice to allow entering just a height and have the width calculated automatically based on the aspect ratio.

**Comment by olafgrabienski** [comment]
I was able to reproduce the bug on a Backdrop 1.12.x test site.

**Comment by BWPanda** [comment]
I just tested this again with two images: one with just the width set, and one with just the height set. It seems that doing this simply sets those attributes in the `<img>` HTML tag. In the case of the height being set, the reason it doesn't work OOTB is that Basis adds some CSS that sets the height to `auto`. Removing that CSS fixes this so the images displays at the correct height.

![image](https://user-images.githubusercontent.com/2385329/151485128-742e01b3-f28d-47fd-86fb-33ab8c24796f.png)

So this isn't a bug after all. A simple CSS override will fix this for anyone that comes across it.

**Comment by olafgrabienski** [comment]
> A simple CSS override will fix this for anyone that comes across it.

That's true for many, but let's keep in mind that not anyone is capable to do a CSS override. Not sure what to do here, though.

**Comment by klonos** [comment]
I also think that there's room for improvement here, so that things work from the UI rather than assuming that content editors would go and edit code.

**Comment by BWPanda** [comment]
I'm not sure what needs to happen here... Setting just a width or height for an image in CKEditor sets that value on the HTML tag. That works, no bug there. And the HTML tag with just the width or the height correctly shows the image proportionately, no bug there either. The only issue is with the Basis theme setting `height: auto;` for image tags (essentially overriding the height). But it does this for responsiveness, which is what we want 98% of the time. On the rare occasion that someone wants to hard-code a height for an image, I think it's reasonable to assume they should then update/override the CSS too...

I don't think there's anything here to change in core. But happy to keep this open and see if/what others think.

**Comment by BWPanda** [comment]
> That's true for many, but let's keep in mind that not anyone is capable to do a CSS override.

If someone wants to set a height in the UI and can't override CSS, then they need to use a different theme me thinks.

**Comment by olafgrabienski** [comment]
> If someone wants to set a height in the UI and can't override CSS, then they need to use a different theme me thinks.

I see, but it remains an issue for less experienced people: If they set (only) the height, they may not realize the problem (or the reason for the problem) and need an experienced site architect to find it and to either override CSS or change the theme. Still not sure what to do. Maybe add help text in the editor dialog which recommends to not set only height?

**Comment by indigoxela** [comment]
A new contrib module might help here to improve UX: [Editor Image Dimension Sync](https://backdropcms.org/project/editorimgdimensionsync).
